pkg/repository/size-validation.go

```diff
@@ -0,0 +1,131 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	adminv2 "github.com/metal-stack/api/go/metalstack/admin/v2"
+	apiv2 "github.com/metal-stack/api/go/metalstack/api/v2"
+	"github.com/metal-stack/metal-apiserver/pkg/db/metal"
+	"github.com/metal-stack/metal-apiserver/pkg/errorutil"
+)
+
+func (r *sizeRepository) validateCreate(ctx context.Context, req *adminv2.SizeServiceCreateRequest) error {
+	var (
+		errs []error
+	)
+
+	err := r.validateSizeConstraints(req.Size.Constraints)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	err = r.validateSizesNotOverlapping(ctx, req.Size)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return errorutil.NewInvalidArgument(errors.Join(errs...))
+	}
+
+	return nil
+}
+
+func (r *sizeRepository) validateUpdate(ctx context.Context, req *adminv2.SizeServiceUpdateRequest, _ *metal.Size) error {
+	var (
+		errs []error
+	)
+
+	err := r.validateSizeConstraints(req.Constraints)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	size := &apiv2.Size{
+		Id:          req.Id,
+		Name:        req.Name,
+		Description: req.Description,
+		Constraints: req.Constraints,
+	}
+
+	err = r.validateSizesNotOverlapping(ctx, size)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return errorutil.NewInvalidArgument(errors.Join(errs...))
+	}
+
+	return nil
+}
+
+func (r *sizeRepository) validateDelete(ctx context.Context, req *metal.Size) error {
+	var errs []error
+
+	// FIXME find all machines with this size
```

Comment:
This can only be fixed once machine service is implemented, right? Should we add an issue so we don't forget?